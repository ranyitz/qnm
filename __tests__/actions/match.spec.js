const matchAction = require('../../src/actions/match');
const { resolveWorkspace } = require('../utils');

describe('match', () => {
  it('should print the matched modules according to passed string', () => {
    const workspace = resolveWorkspace('mix-modules');
    const output = matchAction(workspace, 'anot');

    expect(output).toMatchSnapshot();
  });

  it('should show a message when no module has matched the provided string', () => {
    const workspace = resolveWorkspace('single-module');
    try {
      matchAction(workspace, 'bla');
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });
});
