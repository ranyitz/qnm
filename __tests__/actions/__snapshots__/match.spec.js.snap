// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match should disable colors 1`] = `
"[4m[35manot[39mher[24m
â””â”€â”€ 1.0.0
"
`;

exports[`match should print the matched modules according to passed string 1`] = `
"[4m[35manot[39mher[24m
â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`match should show a message when no module has matched the provided string 1`] = `"Could not find any module that matches \\"bla\\""`;

exports[`match should show why info 1`] = `
"[4m[35mtes[39mt[24m
â””â”€â”€ [37m1.0.0[39m [33m(devDependencies, npm install test)[39m
"
`;
